dlls/winebus.sys/bus_udev.c

```diff
@@ -1701,28 +1701,27 @@ static void udev_add_device(struct udev_device *dev, int fd)
         memcpy(desc.serialnumber, zeros, sizeof(zeros));
     }
 
-    if (!is_hidraw_enabled(desc.vid, desc.pid, axes, buttons))
-    {
-        TRACE("hidraw %s: deferring %s to a different backend\n", debugstr_a(devnode), debugstr_device_desc(&desc));
-        close(fd);
-        return;
-    }
-    if (is_sdl_blacklisted(desc.vid, desc.pid))
-    {
-        /* this device is blacklisted */
-        TRACE("hidraw %s: ignoring %s, in SDL blacklist\n", debugstr_a(devnode), debugstr_device_desc(&desc));
-        close(fd);
-        return;
-    }
 #ifdef HAS_PROPER_INPUT_HEADER
-    else
-        desc.is_gamepad = (axes == 6 && buttons >= 14);
+    desc.is_gamepad = (axes == 6 && buttons >= 14);
 #endif
 
     TRACE("dev %p, node %s, desc %s.\n", dev, debugstr_a(devnode), debugstr_device_desc(&desc));
 
     if (strcmp(subsystem, "hidraw") == 0)
     {
+        if (!is_hidraw_enabled(desc.vid, desc.pid, axes, buttons))
```

Comment:
These `if`s should probably revert `is_gamepad` then in case it has been unconditionally enabled above. Or the above (previous) `else` should be moved lower. `desc.is_gamepad` must not be enabled if a HOTAS was detected, otherwise it enables the xinput driver for such devices which confuses games.

Reply:
Also looking at this again, `is_gamepad` probably still needs proper handling and wasn't working correctly in the first place (just by accident).

Reply:
If `desc` is thrown away after returning from the function, then everything should be fine. I didn't check the call stack, tho. I just was involved here to improve heuristics for HOTAS detection (this is why `is_hidraw_enabled` got additional parameters for axes and buttons). A `hwdb` check could probably placed there.

`is_hidraw_enabled` currently looks at known VID/PID, and additionally checks some well-known VID for button count (many HOTAS have either 31 or 128 buttons, some may have 8 axes but that's currently not used, so if it matches a known VID, it seems good enough as a heuristic).

Reply:
Yup. My reading of the code is `desc` is just a [local variable](https://github.com/ValveSoftware/wine/blob/66cd5c1/dlls/winebus.sys/bus_udev.c#L1617) that is being filled in for a call at the end to `bus_event_queue_device_created(..., ..., &desc)`, so shouldn't have any effects if there is a return before this call.

Reply:
And this on also as resolved?